p2p/net/swarm/swarm_listen.go

```diff
@@ -13,58 +14,56 @@ import (
 	manet "github.com/ipfs/go-ipfs/Godeps/_workspace/src/github.com/jbenet/go-multiaddr-net"
 	ps "github.com/ipfs/go-ipfs/Godeps/_workspace/src/github.com/jbenet/go-peerstream"
 	context "github.com/ipfs/go-ipfs/Godeps/_workspace/src/golang.org/x/net/context"
-	multierr "github.com/ipfs/go-ipfs/thirdparty/multierr"
 )
 
-// Open listeners for each network the swarm should listen on
-func (s *Swarm) listen(addrs []ma.Multiaddr) error {
-
+// Open listeners and reuse-dialers for the given addresses
+func (s *Swarm) setupAddresses(addrs []ma.Multiaddr) error {
```

Comment:
mostly because it sets up dialers at the same time. the idea is that each address passed in is one we can both receive connections on, and make connections from. Could rename it to 'setup transports' instead.